cloudify_agent/shell/commands/configure.py

```diff
@@ -49,8 +45,63 @@ def configure(disable_requiretty, relocated_env, no_sudo):
         click.echo('Disabling requiretty directive in sudoers file')
         _disable_requiretty(no_sudo)
         click.echo('Successfully disabled requiretty for cfy-agent')
-    if relocated_env:
-        _relocate_venv()
+    if fix_shebangs:
+        click.echo('Fixing shebangs in scripts')
+        _fixup_scripts()
+
+
+def _fixup_scripts():
+    """Make scripts in bin_dir relative by rewriting their shebangs
+
+    Examine each file in bin_dir - if it looks like a python script, and has a
+    shebang - replace it with a new, "relative" shebang.
+    """
+    from cloudify_agent.shell.main import get_logger
+    logger = get_logger()
+
+    new_shebang = f"#!{sys.executable}"
+    logger.debug('New shebang = %s', new_shebang)
+    for filename in _find_scripts_to_fix(os.path.dirname(sys.executable)):
+        logger.debug('Making script {0} relative'.format(filename))
+        _rewrite_shebang(filename, new_shebang)
+
+
+def _find_scripts_to_fix(bin_dir):
+    """Search bin_dir for files that look like python scripts with a shebang
+    """
+    for filename in os.listdir(bin_dir):
+        filename = os.path.join(bin_dir, filename)
+        if not os.path.isfile(filename):   # ignore subdirs, e.g. .svn ones
+            continue
+
+        with open(filename, 'rb') as f:
+            try:
+                lines = f.read().decode('utf-8').splitlines()
```

Comment:
This isn't great. For every file we're investigating, we read it in entirely, and then later we read it entirely again.

I'd suggest we either combine the functions so that we read, then immediately rewrite if we should do so; or we modify this so that we only read the first, say, 25 characters.
However, I think rolling the two into one would make more sense- no point opening the file twice.

Reply:
it's now doing `.readline()`. Good.
BTW, I note that shebangs are effectively very limited size (128, by linux kernel hardcoding), and file reads work in terms of disk pages (well... 4096?), so 25 is unnecessarily limiting :)